pkg/service/admission/externalip_admission.go

```diff
@@ -0,0 +1,68 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	kadmission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	apierrs "k8s.io/kubernetes/pkg/api/errors"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+const ExternalIPPluginName = "ExternalIPRanger"
+
+func init() {
+	kadmission.RegisterPlugin("ExternalIPRanger", func(client clientset.Interface, config io.Reader) (kadmission.Interface, error) {
+		return NewExternalIPRanger(nil), nil
+	})
+}
+
+type externalIPRanger struct {
+	*kadmission.Handler
+	network *net.IPNet
+}
+
+var _ kadmission.Interface = &externalIPRanger{}
+
+// NewConstraint creates a new SCC constraint admission plugin.
+func NewExternalIPRanger(network *net.IPNet) *externalIPRanger {
+	return &externalIPRanger{
+		Handler: kadmission.NewHandler(kadmission.Create, kadmission.Update),
+		network: network,
+	}
+}
+
+// Admit determines if the service should be admitted based on the configured network CIDR.
+func (r *externalIPRanger) Admit(a kadmission.Attributes) error {
+	if a.GetResource() != kapi.Resource("services") {
```

Comment:
I don't think you want to match on subresources.  We should really update the `Handler` to do what we want.